WowCharComparerWebApp/Controllers/User/LoginController.cs

```diff
@@ -8,5 +14,86 @@ public IActionResult Index()
         {
             return View();
         }
+
+        [Route("Login/PasswordRecovery")]
+        [HttpGet]
+        public IActionResult PasswordRecoveryView()
+        {
+            return View("PasswordRecovery");
+        }
+
+        [Route("Login/PasswordRecovery/Confirmation")]
+        public IActionResult PasswordResetConfirmationView(string userID)
+        {
+            ViewBag.Message = userID;
+            return View("PasswordChanged");
+        }
+
+        [HttpPost]
+        public IActionResult PasswordChanged(string newPassword, string newPasswordConfirmation, Guid userID)
+        {
+            List<string> passwordsToCheck = new List<string>();
+            {
+                passwordsToCheck.Add(newPassword);
+                passwordsToCheck.Add(newPasswordConfirmation);
+            }
+
+            List<(bool,string)> checkedPasswords = new List<(bool,string)>();
+            RegisterController registerController = new RegisterController();
+
+            for (int index = 0; index < passwordsToCheck.Count(); index++)
+                checkedPasswords.AddRange(registerController.CheckPassword(passwordsToCheck[index]));
+
+            if (checkedPasswords.All(x => x.Item1.Equals(true)))
+            {
+                UpdateUserPassword(newPassword,userID);
+                return Content("Password has been changed");
+                //TODO  *Add generic view to display various messages with same layout* - TASK
+            }
+            else
+            {
+                return Content("Password must be between 8 and 30 characters. Contains one uppercase and one number digit");
+                //TODO  *Add generic view to display various messages with same layout* - TASK
+            }
+        }
+
+        private void UpdateUserPassword(string newPassword, Guid userID)
+        {
+            using (ComparerDatabaseContext db = new ComparerDatabaseContext())
+            using (IDbContextTransaction transaction = db.Database.BeginTransaction())
```

Comment:
No need to use transaction here. Just open db context and after operation simply db.SaveChanges()